shared/util/preconditions.go

```diff
@@ -45,21 +89,42 @@ func ValidateEnvironment(ctx context.Context) error {
 		}
 	}
 
-	_, err = exec.LookPath("aws")
+	return errs.ErrorOrNil()
+}
+
+func ValidateAwsInstalled(ctx context.Context) error {
+	var errs *multierror.Error
+	_, err := exec.LookPath("aws")
 	if err != nil {
 		errs = multierror.Append(errs, errors.Wrap(err, "could not find aws cli in path, run 'brew install awscli' or follow https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html"))
 	}
 
-	_, err = exec.LookPath("terraform")
+	return errs.ErrorOrNil()
+}
+
+func ValidateAwsSessionManagerPluginInstalled(ctx context.Context) error {
```

Comment:
This is an ECS specific tool, we probably need to conditionally validate.

Reply:
I'd say for this pass, just always validate wherever it might be used by a given command. Trying to figure out when it might be required seems potentially fraught. If it really annoys people, we can add that logic later.